admin/client/EditorColorScaleSection.tsx

```diff
@@ -107,21 +119,25 @@ class ColorsSection extends React.Component<{
         this.disposers.push(
             // When the user disables automatic classification,
             // populate with automatic buckets.
+
+            // TODO extract this into a function with a `wipe` param.
+            // Here the `wipe` param is false.
+            // When the user edits a section while in auto mode, `wipe` is true.
             reaction(
-                () => scale.config.isManualBuckets,
+                () => scale.isManualBuckets,
```

Comment:
So I'm not 100% sure about this and this is mostly pre-PR code, but I _think_ reactions modifying observables might be anti-pattern and such reactions should be replaced with `computeds`. The rough reasoning is mobx can build a proper graph with `computed` and do proper state management, whereas `reactions` are asynchronous, untracked, etc. So when you use these kinds of reactions, you can build undetectable cycles in the dependency graph, etc. This is something I've encountered with the Timeline.

Reply:
Yeah I totally agree! And I feel you on the `Timeline` 😄 

I found a way to [get rid of the reaction](https://github.com/owid/owid-grapher/pull/544/commits/7423aa480bc6554dcf6845c507de6af4a3068374) by hiding the "Manual" option.

Reply:
👌 👌 .

Maybe we can do this with the "Color scheme" and "Custom" too, eventually!